tests/cbtest/gtest/test_uzfs.cc

```diff
@@ -114,6 +116,87 @@ uzfs_mgmt_conn_list_count(struct uzfs_mgmt_conn_list *list)
 	return count;
 }
 
+extern int epollfd;
+TEST(uZFS, asyncTaskProps) {
+
+	async_task_t *arg;
+	uzfs_mgmt_conn_t *conn;
+
+
+	conn = SLIST_FIRST(&uzfs_mgmt_conns);
+
+	epollfd = epoll_create1(0);
+	if (epollfd < 0) {
+		perror("epoll_create1");
+		zk_thread_exit();
+		return;
+	}
+
+	uzfs_zinfo_take_refcnt(zinfo, B_TRUE);
+	arg = (async_task_t *)kmem_zalloc(sizeof (*arg), KM_SLEEP);
+	arg->conn = conn;
+	arg->zinfo = zinfo;
+	arg->payload_length = 0;
+	arg->payload = NULL;
+	arg->finished = B_FALSE;
+
+	/* async task should not be processed */
+	mutex_enter(&async_tasks_mtx);
```

Comment:
we don't need to take locks

Reply:
async_tasks is global data structure, multiple threads are can modify this linked list in production code. In Unit testing code there is single thread which can modify it so we can avoid taking lock in
unit testing code. From my personal exprience, I have seen future enahcements/improvement are based on legacy code, I do not  want that tomorrow some developer take this instance as an example and start coding for actual production code, that is why I have taken lock. If you are not agree with this reasoning, I will be more than happy to remove locking code. Please do let me know.